	IOException error(final Throwable why) {
		final IOException e;
		e = new IOException("Encryption error: " + why.getMessage());
		e.initCause(why);
		return e;
	}

